pkg/kubectl/cmd/annotate.go

```diff
@@ -113,7 +113,7 @@ func NewCmdAnnotate(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:     "annotate [--overwrite] (-f FILENAME | TYPE NAME) KEY_1=VAL_1 ... KEY_N=VAL_N [--resource-version=version]",
 		Short:   i18n.T("Update the annotations on a resource"),
-		Long:    annotateLong,
+		Long:    templates.LongDesc(fmt.Sprintf(annotateLong, f.ValidResourcesFromDiscoveryClient())),
```

Comment:
I don't think we should talk to the server to display help. I'd rather see a dedicated command for displaying available resources, and reference that command in this help string

Reply:
I write a proposal for this feature, ptal https://github.com/kubernetes/community/pull/1017